logprep/processor/generic_adder/mysql_connector.py

```diff
@@ -0,0 +1,140 @@
+"""This module is used to connect to a MySQL database and to retrieve data from a SQL table."""
+
+import time
+from logging import Logger
+
+import mysql.connector as db
+
+
+class MySQLConnector:
+    """Used to connect to a MySQL database and to retrieve data from a table if it has changed."""
+
+    def __init__(self, sql_config: dict, logger: Logger):
+        """Initialize the MySQLConnector.
+
+        Parameters
+        ----------
+        sql_config : dict
+           SQL configuration dictionary.
+        logger : logging.Logger
+           Logger to use.
+
+        Returns
+        -------
+        bool
+            True if the SQL table has changed, False otherwise.
+
+        """
+        self._logger = logger
+
+        self.con = db.connect(
+            user=sql_config["user"],
+            password=sql_config["password"],
+            host=sql_config["host"],
+            database=sql_config["database"],
+            port=sql_config.get("port", 3306),
+        )
+
+        self.cur = self.con.cursor()
+
+        self.target_column = sql_config["target_column"]
+        self._add_target_column = sql_config.get("add_target_column", False)
+
+        self.table_name = sql_config["table"]
+
+        self._timer = sql_config.get("timer", 60 * 3)
+        self._last_check = 0
+        self._last_table_checksum = None
+
+    def check_change(self) -> bool:
+        """Check if a configured SQL table has changed.
+
+        The checksum of the table is used to check if a table has changed. The check is only
+        performed if a specified time has passed since the last check.
+
+        Returns
+        -------
+        bool
+            True if the SQL table has changed, False otherwise.
+
+        """
+        if time.time() - self._last_check >= self._timer:
+            self._last_check = time.time()
+            checksum = self._get_checksum()
+            if self._last_table_checksum is None:
+                self._last_table_checksum = checksum
+                return True
+            if self._last_table_checksum == checksum:
+                return False
+            return True
+        return False
+
+    def _get_checksum(self) -> int:
+        """Get the checksum a configured SQL table.
+
+        The checksum is used to check if a table has changed.
+
+        Returns
+        -------
+        int
+            The checksum of a SQL table.
+
+            This value changes if the table or it's contents change.
+
+        """
+        self.cur.execute(f"CHECKSUM TABLE {self.table_name}")  # nosemgrep
```

Comment:
I'm not sure how this sql courser works, but is it possible to make an SQL-Infection here by adding some SQL-Statement in the config field of the `table_name`? Maybe there is a good way to escape the `table_name` first. If this works we can maybe also remove the `# nosemgrep` again.